frontend/public/src/containers/pages/checkout/CartPage.js

```diff
@@ -154,6 +155,20 @@ export class CartPage extends React.Component<Props, CartState> {
     )
   }
 
+  renderFinancialAssistanceOffer() {
+    const { cartItems, discounts } = this.props
+    const userFlexiblePriceExists = discounts && discounts.length > 0 ? discounts.find(discount => discount.for_flexible_pricing === true) : false
```

Comment:
This seems to return false whenever a discount is applied, whether or not it's flagged for flexible pricing.